src/main/groovy/com/netflix/gradle/plugins/utils/GradleUtils.groovy

```diff
@@ -41,6 +43,13 @@ final class GradleUtils {
             File sourceRoot = new File("/$sourceBase", sourceRelative)
             File targetRoot = new File("/$sourceBase", targetPath.substring(sourceBasePath.length()))
             Path relativeTarget = sourceRoot.isDirectory() ? sourceRoot.toPath().relativize(targetRoot.toPath()) : sourceRoot.parentFile.toPath().relativize(targetRoot.toPath())
```

Comment:
Interestingly enough, if I just take the fork out of this line and leave the following all tests pass
```
Path relativeTarget = sourceRoot.parentFile.toPath().relativize(targetRoot.toPath())
```
It still doesn't answer _why_ the link is being evaluated differently, but I don't ever see a use case for the first option here anyways.  I'm probably missing something, but whenever this is successful, it evaluates false.  Again, I believe that is correct since it is a link and isDirectory() should return false for that.

Even if we are at a "directory symlink", we would still want the relative path to the target from the _parent_ I believe.  Otherwise the path you get is going up out of the "link directory" and back down to the target. Like `../path/to/target`.  Then the link is broken.